platform/posix/transport/src/openssl_posix.c

```diff
@@ -746,13 +746,17 @@ int32_t Openssl_Recv( NetworkContext_t * pNetworkContext,
         /* Set the file descriptor for poll. */
         pollFds.fd = pOpensslParams->socketDescriptor;
 
+        /* #SSL_pending returns a value > 0 if application data
+         * from the last processed TLS record remains to be read. */
         if( ( bytesToRecv > 1 ) || ( SSL_pending( pOpensslParams->pSsl ) > 0 ) )
```

Comment:
It could be fine to call `SSL_pending` for every call to this function, irrespective of the number of bytes requested. This can act as the first level of filter where the OpenSSL library is queried about pending data, followed by checking the `poll` function for data from the TCP socket